### PR TITLE
Add Discord.FR

### PR DESCRIPTION
**Pull request by PixelNotix**
Hello ! 
I would like to add the presence for the https://discord.fr website.
Here is my discord ID: 552550090796171265

### CODE REVIEW

**Comment by PixelNotix** [comment]
> **Screenshots of working**

> HomePage	homepage
> Wiki	wiki
> Blog	blog
> Server page 	server
> Forms 1	forms1
> Forms 2	forms2
> Forms 3	forms3
> Privacy 	privacy
> Staff	staff
> Status	status
> Support	support

File: websites/D/Discord FR/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    "staff.discord.fr"
+  ],
+    "description": {
+      "en": "Network bringing together the francophone Discord community. Wiki, blog, and tutorials in French. Active, attentive, and experienced staff, ready to help you configure your accounts, servers, and bots. Dive into an entirely francophone world and bring your French to the next level!",
+      "fr": "Réseau regroupant la communauté Discord francophone. Wiki, blog et tutoriels en français. Un staff actif, attentif et expérimenté, prêt à vous aider pour configurer vos comptes, serveurs et bots. Rencontre la première communauté française bénévole d'entraide en france !"
+    },
+    "service": "Discord FR",
+    "version": "1.0.0",
+    "logo": "https://i.discord.fr/Kd3.jpg",
+    "thumbnail": "https://i.discord.fr/nmj.jpg",
```
Comment (Bas950) [inline websites/D/Discord FR/dist/metadata.json:38]: upload to imgur

File: websites/D/Discord FR/presence.ts [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+  It is recommended to set up another function outside of this event function which will change variable values and do the heavy lifting if you call data from an API.*/
+
+  if (presenceData.details == null) {
+      //This will fire if the presence details are not set
+      presence.setTrayTitle(); //Clears the tray title for mac users
+      presence.setActivity(); /*Update the presence with no data, therefore clearing it and making the large image the Discord Application icon, and the text the Discord Application name*/
+  } else {
+      //This will fire when the presence details are set
+      presence.setActivity(presenceData); //Update the presence with all the values from the presenceData object
+  }
```
Comment (Bas950) [inline websites/D/Discord FR/presence.ts:231]: remove comments

**Review by Bas950 — CHANGES_REQUESTED** [review]
Fix codacy issues

File: websites/D/Discord FR/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      },
+      {
+        "name": "Zallom",
+        "id": "268108606027726849"
+      },
+      {
+        "name": "Ichii",
+        "id": "530061924130488351"
+      }
+    ],
```
Comment (ririxidev) [inline websites/D/Discord FR/dist/metadata.json:20]: delete contributors.

File: websites/D/Discord FR/presence.ts [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+      presenceData.details = "Erreur lors du chargement";
+      presenceData.state = "support.discord.fr";
+    }
+
+  }
+
+}
+
+updatePresenceData();
+setInterval(updatePresenceData, 10000);
```
Comment (ririxidev) [inline websites/D/Discord FR/presence.ts:218]: why? xD

File: websites/D/Discord FR/presence.ts [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+let presenceData: PresenceData = {
+  largeImageKey: "discordfr",
+  smallImageKey: "rooster",
+  details: "Chargement de la page...",
+  state: "Discord FR",
+  startTimestamp: timestamp
+};
+
+var title: any;
```
Comment (ririxidev) [inline websites/D/Discord FR/presence.ts:20]: no vars

File: websites/D/Discord FR/presence.ts [hunk 6]
```diff
@@ -0,0 +1,232 @@
+let presence = new Presence({
```
Comment (ririxidev) [inline websites/D/Discord FR/presence.ts:1]: const

File: websites/D/Discord FR/presence.ts [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+  clientId: "563836644579606528" // DFR CLIENT ID
+}),
+
+strings = presence.getStrings({
+  play: "presence.playback.playing",
+  pause: "presence.playback.paused"
+});
+
+let timestamp = Math.floor(Date.now() / 1000);
```
Comment (ririxidev) [inline websites/D/Discord FR/presence.ts:10]: const

File: websites/D/Discord FR/presence.ts [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+  largeImageKey: "discordfr",
+  smallImageKey: "rooster",
+  details: "Chargement de la page...",
+  state: "Discord FR",
+  startTimestamp: timestamp
+};
+
+var title: any;
+const ticket = document.location.pathname.split("/");
+const search = document.location.pathname.split("/?q=");
```
Comment (ririxidev) [inline websites/D/Discord FR/presence.ts:22]: merge together

File: websites/D/Discord FR/presence.ts [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+}),
+
+strings = presence.getStrings({
+  play: "presence.playback.playing",
+  pause: "presence.playback.paused"
+});
+
+let timestamp = Math.floor(Date.now() / 1000);
+
+let presenceData: PresenceData = {
```
Comment (ririxidev) [inline websites/D/Discord FR/presence.ts:12]: const and merge with other...

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/D/Discord FR/presence.ts [hunk 10]
```diff
@@ -0,0 +1,191 @@
+const presence = new Presence({
+  clientId: "563836644579606528"
+});
+
+const timestamp = Math.floor(Date.now() / 1000);
+
+const presenceData: PresenceData = {
```
Comment (ririxidev) [inline websites/D/Discord FR/presence.ts:7]: merge these consts

File: websites/D/Discord FR/presence.ts [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+  largeImageKey: "discordfr",
+  smallImageKey: "rooster",
+  details: "Chargement de la page...",
+  state: "Discord FR",
+  startTimestamp: timestamp
+};
+
+let title: any;
```
Comment (ririxidev) [inline websites/D/Discord FR/presence.ts:15]: no any type 😉

File: websites/D/Discord FR/presence.ts [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+      presenceData.state = "support.discord.fr";
+    }
+
+  }
+  else {
+    presence.setActivity();
+    presence.setTrayTitle();
+  }
+
+  updatePresenceData();
```
Comment (ririxidev) [inline websites/D/Discord FR/presence.ts:190]: still why ;-; please use ``presence.on("UpdateData", async () => { });`` pls not different function bruh.

**Review by ririxidev — CHANGES_REQUESTED** [review]

**Comment by Isa27-0** [comment]
Sorry to disturb you, but changing the large image and the small image on each page not go against Discord guidelines for presence?

**Comment by ririxidev** [comment]
They don't really have guidelines for it, but yea, maybe really remove changing the largeImageKey on every webpage.

**Review by Bas950 — CHANGES_REQUESTED** [review]
Not even an updatedata function...

**Comment by ririxidev** [comment]
no response from pull request author.

**Pull request closed without merging**